api/main_endpoints/routes/OfficeAccessCard.js

```diff
@@ -0,0 +1,84 @@
+const {
+  UNAUTHORIZED,
+  BAD_REQUEST,
+  SERVER_ERROR,
+  NOT_FOUND,
+  OK,
+} = require('../../util/constants').STATUS_CODES;
+const express = require('express');
+const router = express.Router();
+const bodyParser = require('body-parser');
+const OfficeAccessCard = require('../models/OfficeAccessCard.js');
+const logger = require('../../util/logger');
+const { officeAccessCard = {} } = require('../../config/config.json');
+const { API_KEY = 'NOTHING_REALLY' } = officeAccessCard;
+
+
+router.use(bodyParser.json());
+
+function checkIfCardExists(cardBytes) {
+  return new Promise((resolve) => {
+    try {
+      OfficeAccessCard.findOne({ cardBytes }, (error, result) => {
+        if (error) {
+          logger.error('checkIfCardExists got an error querying mongodb: ', error);
+          return resolve(false);
+        }
+        return resolve(!!result);
+      });
+    } catch (error) {
+      logger.error('checkIfCardExists caught an error: ', error);
+      return resolve(false);
+    }
+  });
+}
```

Comment:
```suggestion
async function checkIfCardExists(cardBytes) {
  const result = await OfficeAccessCard.findOne({ cardBytes }).exec().catch(error => {
    logger.error('checkIfCardExists got an error querying mongodb: ', error);
    return null;
  });
  return !!result;
}
```

Probably a lot cleaner w/ async await 

https://mongoosejs.com/docs/5.x/docs/api.html#model_Model.findOne

Reply:
the above suggestion is using both async/await and promise chaining

Reply:
i will try using
```js
try {
  return !!await OfficeAccessCard.findOne({ cardBytes });
} catch (e) {
  logger.error('checkIfCardExists got an error querying mongodb: ', error);
}
```

but i will wait to test until someone is able to scan a card in the room to verify that this works

Reply:
common enough to be in the [mozilla docs](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/async_function#rewriting_a_promise_chain_with_an_async_function)

write it with a try/catch if you want